mne/preprocessing/tests/test_regress.py

```diff
@@ -34,5 +35,90 @@ def test_regress_artifact():
     assert orig_norm / 2 > clean_norm > orig_norm / 10
     with pytest.raises(ValueError, match=r'Invalid value.*betas\.shape.*'):
         regress_artifact(epochs, betas=betas[:-1])
-    with pytest.raises(ValueError, match='cannot be contained in'):
-        regress_artifact(epochs, picks='eog', picks_artifact='eog')
+    # Regressing channels onto themselves should work
+    epochs, betas = regress_artifact(epochs, picks='eog', picks_artifact='eog')
+    assert np.ptp(epochs.get_data('eog')) < 1E-15  # constant value
+    assert_allclose(betas, 1)
+
+    # Test applying TAA.
+    raw_no_taa, betas = regress_artifact(raw, taa=False)
+    raw_taa, _ = regress_artifact(raw, taa=True, betas=betas)
+    # TAA should have increased signal amplitude
+    assert np.ptp(raw_no_taa.get_data('eeg')) < np.ptp(raw_taa.get_data('eeg'))
+
+
+@testing.requires_testing_data
+def test_eog_regression():
+    """Test regressing artifact data using the EOGRegression class."""
+    raw = read_raw_fif(raw_fname).pick(['eeg', 'eog', 'stim'])
+
+    # Test error regarding preloading
+    with pytest.raises(RuntimeError, match='requires raw data to be loaded'):
+        model = EOGRegression().fit(raw)
+
+    raw.load_data()
+
+    # Test regression on raw data
+    model = EOGRegression().fit(raw)
+    assert model.coef_.shape == (59, 1)  # 59 EEG channels, 1 EOG channel
+    raw_clean = model.apply(raw)
+    # Some signal must have been removed
+    assert np.ptp(raw_clean.get_data('eeg')) < np.ptp(raw.get_data('eeg'))
```

Comment:
shouldn't it be true that
```py
assert np.all(np.ptp(raw_clean.get_data('eeg'), axis=-1) < np.ptp(raw.get_data('eeg'), axis=-1))
```
?  I.e., peak-to-peak is reduced *in each channel*?  That seems like a more stringent test

Reply:
Good catch!